tests/nlu/classifiers/test_diet_classifier.py

```diff
@@ -812,8 +812,171 @@ def test_removing_label_sparse_feature_sizes(
+        {
+            "name": "CountVectorsFeaturizer",
+            "analyzer": "char_wb",
+            "min_ngram": 1,
+            "max_ngram": 4,
+        },
+        {"name": "DIETClassifier", EPOCHS: 1},
+    ]
+    _config = RasaNLUModelConfig({"pipeline": pipeline, "language": "en"})
+
+    (_, trained, persisted_path) = await rasa.nlu.train.train(
+        _config,
+        path=str(tmpdir),
+        data="data/test_incremental_training/iter1/",
+        component_builder=component_builder,
+    )
+    assert trained.pipeline
+    message = Message.build(text="Rasa is great!")
+    trained.featurize_message(message)
+    old_sparse_feature_sizes = message.get_sparse_feature_sizes(attribute=TEXT)
+    initial_diet_layers = (
+        trained.pipeline[-1]
+        .model._tf_layers["sequence_layer.text"]
+        ._tf_layers["feature_combining"]
+    )
+    initial_diet_sequence_layer = initial_diet_layers._tf_layers[
+        "sparse_dense.sequence"
+    ]._tf_layers["sparse_to_dense"]
+    initial_diet_sentence_layer = initial_diet_layers._tf_layers[
+        "sparse_dense.sentence"
+    ]._tf_layers["sparse_to_dense"]
+
+    initial_diet_sequence_size = initial_diet_sequence_layer.get_kernel().shape[0]
+    initial_diet_sentence_size = initial_diet_sentence_layer.get_kernel().shape[0]
+    assert initial_diet_sequence_size == sum(
+        old_sparse_feature_sizes[FEATURE_TYPE_SEQUENCE]
+    )
+    assert initial_diet_sentence_size == sum(
+        old_sparse_feature_sizes[FEATURE_TYPE_SENTENCE]
+    )
+
+    loaded = Interpreter.load(persisted_path, component_builder, new_config=_config,)
+    assert loaded.pipeline
+    assert loaded.parse("Rasa is great!") == trained.parse("Rasa is great!")
+    (_, trained, _) = await rasa.nlu.train.train(
+        _config,
+        path=str(tmpdir),
+        data="data/test_incremental_training/",
+        component_builder=component_builder,
+        model_to_finetune=loaded,
+    )
+    assert trained.pipeline
+    message = Message.build(text="Rasa is great!")
+    trained.featurize_message(message)
+    new_sparse_feature_sizes = message.get_sparse_feature_sizes(attribute=TEXT)
+
+    final_diet_layers = (
+        trained.pipeline[-1]
+        .model._tf_layers["sequence_layer.text"]
+        ._tf_layers["feature_combining"]
+    )
+    final_diet_sequence_layer = final_diet_layers._tf_layers[
+        "sparse_dense.sequence"
+    ]._tf_layers["sparse_to_dense"]
+    final_diet_sentence_layer = final_diet_layers._tf_layers[
+        "sparse_dense.sentence"
+    ]._tf_layers["sparse_to_dense"]
+
+    final_diet_sequence_size = final_diet_sequence_layer.get_kernel().shape[0]
+    final_diet_sentence_size = final_diet_sentence_layer.get_kernel().shape[0]
+    assert final_diet_sequence_size == sum(
+        new_sparse_feature_sizes[FEATURE_TYPE_SEQUENCE]
+    )
+    assert final_diet_sentence_size == sum(
+        new_sparse_feature_sizes[FEATURE_TYPE_SENTENCE]
+    )
```

Comment:
I see that none of the tests check the updated data signatures. Please assert them as well with expected values.